app/assets/javascripts/components/gwas_analysis.coffee

```diff
@@ -69,9 +84,24 @@ window.GwasAnalysis = class GwasAnalysis extends Component
             $errors.find("ul").append($li)
           )
 
-    @$(".delete-#{field}").on 'ajax:success', (data, status, xhr) =>
+    this.$(".delete-#{field}").on 'ajax:success', (data, status, xhr) =>
       $fileinput.removeClass('hidden')
       $result.addClass('hidden')
 
       options.delete && options.delete()
 
+  clearDataFile: (field) =>
+    $fileinput = this.$(".#{field}-fileinput")
+    $result = this.$("div.#{field}")
+
+    $fileinput.removeClass('hidden')
+    $result.addClass('hidden')
+
+    this.$("#analysis_#{field.replace(/-/g, '_')}_id").val("")
+
+  updateGenotypeTemplateLink: (plant_trial_id) =>
+    $link = this.$("#analysis_gwas_genotype_template_download")
+    url = $link.attr("href")
+    url = url.replace(/&plant_trial_id=.*/, '') + "&plant_trial_id=" + plant_trial_id
```

Comment:
This will not work if `plant_trial_id` is the first parameter and it will remove all parameters after that one. Although I see that there is no easy way to do that any better. Can you try to match `plant_trial_id` only and its value up to `&` or end of string?